warehouse/query-core/src/main/java/datawave/query/composite/CompositeTerm.java

```diff
@@ -85,7 +89,21 @@ public void getNodesAndExpressions(List<Class<? extends JexlNode>> nodeClasses,
      * @return
      */
     private String getAppendedExpressions() {
-        return String.join(separator, expressionList);
+        
+        final List<String> newExpressions;
+        if (Iterables.getLast(jexlNodeList) instanceof ASTERNode && !jexlNodeList.stream().allMatch(x -> x instanceof ASTERNode)) {
+            // We have an ASTERNode in the last position, but the rest are not ERNodes ( and therefore must be EQNodes )
+            // escape all of the terms
+            newExpressions = expressionList.stream().map(term -> {
+                return Pattern.quote(term);
+            }).collect(Collectors.toList());
+            // set the last ( the regex ) so that it is not escaped.
+            newExpressions.set(expressionList.size() - 1, expressionList.get(expressionList.size() - 1));
+            
+        } else {
+            newExpressions = expressionList;
+        }
+        return String.join(separator, newExpressions);
```

Comment:
If the composite index has more than 2 terms then we could have something like eq, er, eq. Seems like we want to escape only the first eq nodes until we hit an er node, which is my suggestion below. Though now that I'm typing this, I wonder if we want to escape all eq node terms regardless of position if there's at least one eq and one er node in the list.
```suggestion
        final List<String> newExpressions;
        final JexlNode firstNode = jexlNodeList.get(0);
        final long regexCount = jexlNodeList.stream().filter(x -> x instanceof ASTERNode).count();
        // If the list of terms has a mix of regex and non-regex terms, then we want to escape the
        // non-regex terms until we hit the first regex term. From the first regex on, we don't need
        // to escape.
        if (regexCount == 0 || regexCount == jexlNodeList.size() || (firstNode instanceof ASTERNode)) {
            newExpressions = expressionList;
        } else {
            int regexPos = 0;
            newExpressions = new ArrayList<>(jexlNodeList.size());
            for ( ; regexPos < jexlNodeList.size(); regexPos++) {
                if (jexlNodeList.get(regexPos) instanceof ASTERNode)
                    break;
                newExpressions.add(Pattern.quote(expressionList.get(regexPos)));
            }
            newExpressions.addAll(expressionList.subList(regexPos, expressionList.size()));
        }
        return String.join(separator, newExpressions);
```

Reply:
As per above I'm going to keep this until I can find test the assumptions based on the code that wraps this...